executor/api/grpc/kfserving/client.go

```diff
@@ -0,0 +1,187 @@
+package kfserving
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"github.com/seldonio/seldon-core/executor/api/client"
+	grpc2 "github.com/seldonio/seldon-core/executor/api/grpc"
+	"github.com/seldonio/seldon-core/executor/api/grpc/kfserving/inference"
+	"github.com/seldonio/seldon-core/executor/api/payload"
+	v1 "github.com/seldonio/seldon-core/operator/apis/machinelearning.seldon.io/v1"
+	"google.golang.org/grpc"
+	"io"
+	"math"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+type KFServingGrpcClient struct {
+	Log            logr.Logger
+	callOptions    []grpc.CallOption
+	conns          map[string]*grpc.ClientConn
+	Predictor      *v1.PredictorSpec
+	DeploymentName string
+	annotations    map[string]string
+}
+
+func (s *KFServingGrpcClient) ModelMetadata(ctx context.Context, modelName string, host string, port int32, msg payload.SeldonPayload, meta map[string][]string) (payload.ModelMetadata, error) {
+	panic("implement me")
+}
+
+func NewKFServingGrpcClient(predictor *v1.PredictorSpec, deploymentName string, annotations map[string]string) client.SeldonApiClient {
+	opts := []grpc.CallOption{
+		grpc.MaxCallSendMsgSize(math.MaxInt32),
+		grpc.MaxCallRecvMsgSize(math.MaxInt32),
+	}
+	smgc := KFServingGrpcClient{
+		Log:            logf.Log.WithName("SeldonGrpcClient"),
+		callOptions:    opts,
+		conns:          make(map[string]*grpc.ClientConn),
+		Predictor:      predictor,
+		DeploymentName: deploymentName,
+		annotations:    annotations,
+	}
+	return &smgc
+}
+
+func (s *KFServingGrpcClient) getConnection(host string, port int32, modelName string) (*grpc.ClientConn, error) {
+	k := fmt.Sprintf("%s:%d", host, port)
+	if conn, ok := s.conns[k]; ok {
+		return conn, nil
+	} else {
+		opts := []grpc.DialOption{
+			grpc.WithInsecure(),
+		}
+		opts = append(opts, grpc2.AddClientInterceptors(s.Predictor, s.DeploymentName, modelName, s.annotations, s.Log))
+		conn, err := grpc.Dial(fmt.Sprintf("%s:%d", host, port), opts...)
```

Comment:
Seems first iteration won't have ssl, which is reasonable - but may be worth mentioning briefly on docs or failing in the operator if ssl is enabled with protocl kfserving

Reply:
I don't see SSL on any of the grpc clients.  Sounds like this is a wider issue to address?

Reply:
Yeah sounds good, I think we can just open a gh issue